### PR TITLE
Handle block states optional in <1.19.4

### PR DESCRIPTION
**Pull request by FlorianMichael**
This PR fixes block state reading in versions below 1.19.4 where there was only a block state type which is optional, we can probably also rename some of the meta type fields with ViaVersion 5.

1.19.3 code:
image

1.19.4 code:
image

### CODE REVIEW

File: common/src/main/java/com/viaversion/viaversion/protocols/protocol1_18to1_17_1/packets/EntityPackets.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ protected void registerRewrites() {
             }
         });
 
-        registerMetaTypeHandler(Types1_18.META_TYPES.itemType, null, null, null);
+        registerMetaTypeHandler(Types1_18.META_TYPES.itemType,null, null);
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viaversion/protocols/protocol1_18to1_17_1/packets/EntityPackets.java:97]: Missing space

File: common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/metadata/MetadataRewriter1_13To1_12_2.java [hunk 2]
```diff
@@ -40,7 +40,11 @@ public MetadataRewriter1_13To1_12_2(Protocol1_13To1_12_2 protocol) {
     protected void registerRewrites() {
         filter().mapMetaType(typeId -> Types1_13.META_TYPES.byId(typeId > 4 ? typeId + 1 : typeId));
         filter().metaType(Types1_13.META_TYPES.itemType).handler(((event, meta) -> protocol.getItemRewriter().handleItemToClient(meta.value())));
-        filter().metaType(Types1_13.META_TYPES.blockStateType).handler(((event, meta) -> meta.setValue(WorldPackets.toNewId(meta.value()))));
+        filter().metaType(Types1_13.META_TYPES.blockStateType).handler(((event, meta) -> {
+            if ((int) meta.value() != 0) {
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/metadata/MetadataRewriter1_13To1_12_2.java:44]: put meta.value into a variable

**Review by kennytv — APPROVED** [review]
Accepted because it's cleaner, but this doesn't actually "fix" anything, 0 is always 0

**Comment by FlorianMichael** [comment]
"Fix" is the wrong word, but why would we handle and/or name types wrong, it's just cleaner and prevents confusion when looking at old code.

File: common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/metadata/MetadataRewriter1_13To1_12_2.java (hunk 2 above)
Comment (FlorianMichael) [reply common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/metadata/MetadataRewriter1_13To1_12_2.java]: Fixed

**Pull request merged**